javascript/ql/src/Security/CWE-116/IncompleteSanitization.ql

```diff
@@ -101,6 +101,40 @@ predicate allBackslashesEscaped(DataFlow::Node nd) {
   allBackslashesEscaped(nd.getAPredecessor())
 }
 
+/**
+ * Holds if `repl` looks like a call to "String.prototype.replace" that removes the first occurrence of `str`.
+ */
+predicate removesFirstOccurence(MethodCallExpr repl, string str) {
+  repl.getMethodName() = "replace" and
+  repl.getArgument(0).getStringValue() = str and
```

Comment:
Use `getAMatchedString` here to handle the regex case?

Reply:
That is a deliberate limitation. I think it is more likely that the global flag was forgotten if this is a regex case, and I can not recall any FPs due do regexes.

Reply:
OK, makes sense.